### DISCUSSION TITLE
Best practices for multilabeling

### DISCUSSION THREAD

**Discussion by annarrrzzz** [MaartenGr/BERTopic#2131] [Q&A] [answered]
Hi BERTopic Experts, I have some documents, and would like to cluster them into multiple groups if relevant. Per [this post](https://github.com/MaartenGr/BERTopic/pull/840#issue-1449408747), we can find out the distribution of multiple topics for each document using Topic Distribution. However, per this [other post](https://github.com/MaartenGr/BERTopic/issues/295), "there is a difference between [a topic] having a high probability and actually being contained in the document". How do we consolidate these? More specifically:

1/ Is Topic Distribution the solution to unsupervised multilabeling?
2/ If so, is it advised that we pick the top n topics where the probabilities do not differ by X ppt?

**Comment by MaartenGr** [comment, marked as answer]
> 1/ Is Topic Distribution the solution to unsupervised multilabeling?
 
It is one of the solutions you could indeed take. The most straightforward method (and perhaps most accurate) is to split up your document into sentences instead before feeding them to BERTopic. 

> 2/ If so, is it advised that we pick the top n topics where the probabilities do not differ by X ppt?

There isn't a single answer to this as it highly depends on the data but also your use case. However, there is the `min_similarity` parameter in [`.approximate_distribution`](https://maartengr.github.io/BERTopic/api/bertopic.html#bertopic._bertopic.BERTopic.approximate_distribution) that you can use to decide how similar given similarity scores should be.
